meson.build

```diff
@@ -0,0 +1,231 @@
+cdata.set_quoted('PACKAGE', 'gst-shark', description: 'Name of package')
+cdata.set_quoted('PLUGINDIR', plugins_install_dir, description: 'Directory where plugins are located')
+cdata.set_quoted('VERSION', gst_shark_version, description: 'Version number of package')
+
+if gst_shark_version_nano > 0
+    # Have GST_ERROR message printed when running from git
+    cdata.set('GST_LEVEL_DEFAULT', 'GST_LEVEL_ERROR', description : 'Default errorlevel to use')
+else
+    cdata.set('GST_LEVEL_DEFAULT', 'GST_LEVEL_NONE', description : 'Default errorlevel to use')
+endif
+
+cdata.set('GST_SHARK_VERSION_MAJOR', gst_shark_version_major, description: 'GstShark Version Major')
+cdata.set('GST_SHARK_VERSION_MINOR', gst_shark_version_minor, description: 'GstShark Version Minor')
+cdata.set('GST_SHARK_VERSION_MICRO', gst_shark_version_micro, description: 'GstShark Version Micro')
+cdata.set('GST_SHARK_VERSION_NANO', gst_shark_version_nano, description: 'GstShark Version Nano')
+
+# GStreamer package name and origin url
+gst_package_name = get_option('package-name')
+if gst_package_name == ''
+  if gst_shark_version_nano == 0
+    gst_package_name = 'GstShark source release'
+  elif gst_shark_version_nano == 1
+    gst_package_name = 'GStreamer git'
+  else
+    gst_package_name = 'GStreamer prerelease'
+  endif
+endif
+cdata.set_quoted('GST_PACKAGE_NAME', gst_package_name)
+cdata.set_quoted('GST_PACKAGE_ORIGIN', get_option('package-origin'))
+
+cdata.set('GST_STABLE_RELEASE', 1, description: 'Define this to use the tracers API of GStreamer version >= 1.8.1')
+
+# Checking the OS to enable reading of CPU usage
+if host_machine.system().to_lower() == 'linux'
+    cdata.set('GST_CPUUSAGE_ENABLE', 1, description : 'Define this to enable the tracer for reading of the CPU usage')
+    message('Your platform supports CPU Usage tracing')
+else
+    warning('CPU Usage tracer currently not supported on your platform')
+endif
+
+# Define LIBDIR, GST_DATADIR so we can inform people where we live
+cdata.set_quoted('LIBDIR', join_paths(get_option('prefix'), get_option('libdir')), description: 'library dir')
+cdata.set_quoted('GST_SHARK_DATADIR', join_paths(get_option('prefix'), get_option('datadir')), description: 'data dir')
+
+# These are only needed/used by the ABI tests
+host_defines = [
+  [ 'x86', 'HAVE_CPU_I386' ],
+  [ 'x86_64', 'HAVE_CPU_X86_64' ],
+  [ 'arm', 'HAVE_CPU_ARM' ],
+  [ 'aarch64', 'HAVE_CPU_AARCH64' ],
+  [ 'mips', 'HAVE_CPU_MIPS' ],
+  [ 'powerpc', 'HAVE_CPU_PPC' ],
+  [ 'powerpc64', 'HAVE_CPU_PPC64' ],
+  [ 'alpha', 'HAVE_CPU_ALPHA' ],
+  [ 'sparc', 'HAVE_CPU_SPARC' ],
+  [ 'ia64', 'HAVE_CPU_IA64' ],
+  [ 'hppa', 'HAVE_CPU_HPPA' ],
+  [ 'm68k', 'HAVE_CPU_M68K' ],
+  [ 's390', 'HAVE_CPU_S390' ],
+]
+foreach h : host_defines
+  if h.get(0) == host_machine.cpu()
+    cdata.set(h.get(1), 1)
+  else
+    cdata.set(h.get(1), false, description : 'Define if the target CPU is ' + h.get(0).to_upper())
+  endif
```

Comment:
Could this be simplified as `cdata.set10(h.get(1), h.get(0) == host_machine.cpu())`?

Reply:
I simplified it using only set, thanks, so to only define our cpu